devel/python/python/ert/job_queue/job.py

```diff
@@ -17,27 +17,30 @@
 
 import time
 import datetime
-from ert.cwrap.cclass import CClass
+from ert.cwrap import BaseCClass
 
-# Enum values nicked from libjob_queue/src/basic_queue_driver.h
-STATUS_PENDING = 16
-STATUS_RUNNING = 32
-STATUS_DONE = 64
-STATUS_EXIT = 128
 
+# The wrapping here and the interplay with the queue class is on a
+# 'barely working' standard.
 
-class Job(CClass):
+class Job(BaseCClass):
+    TYPE_NAME = "job"
+    
     def __init__(self, driver, c_ptr, queue_index, blocking=False):
         self.driver = driver
-        self.init_cobj(c_ptr, self.driver.free_job)
         self.submit_time = datetime.datetime.now()
         self.queue_index = queue_index
+        super(Job , self).__init__( c_ptr )
+
+
+    def free(self):
+        pass
 
 
     def block( self ):
         while True:
             status = self.status()
-            if status == STATUS_DONE or status == STATUS_EXIT:
+            if status == JobStatusType.STATUS_DONE or status == JobStatusType.STATUS_EXIT:
```

Comment:
Will fail without an import of `JobStatusType`. I am afraid that no test are directly testing the features in `Job`